force-app/main/default/classes/AccountFieldsSet.cls

```diff
@@ -0,0 +1,25 @@
+public with sharing class AccountFieldsSet {
+    public static void handleInsert(List<Account> newAccounts){
+        List<Account> accountsToBeInserted = new List<Account>();
+        for (Account acc : newAccounts){
+            if (acc.Type == null) {
+                acc.Type = 'Prospect';
+                accountsToBeInserted.add(acc);
+            }
```

Comment:
No need to add to a collection there is no DML needed for this in a before insert trigger.

Reply:
Thanks Brooks. I got to learn this in Trigger Best Practices home work that there is no necessity of including the DML for before update and before insert triggers and we just need to include the changes that we would want to have.